src/faillog.c

```diff
@@ -429,7 +429,7 @@ static bool set_locktime_one (uid_t uid, long locktime)
 	struct faillog fl;
 
 	offset = (off_t) uid * sizeof (fl);
-	if (offset + sizeof (fl) <= statbuf.st_size) {
+	if (offset + sizeof (fl) <= (size_t)statbuf.st_size) {
```

Comment:
You might consider just ignoring the warning, and let the code be smaller to read.  This is a place where default integer conversions are what we want, so I'd say the warning is spurious in this case (a false positive).

However, you may want to silence it to be able to notice non-spurious ones, if they exist.  I've long had a dilemma about this warning, and so far my attitude to it was to not add casts, and instead just ignore the warning (`-Wno-error=sign-compare`).  I'm still interested in the warning, but not in the error.  What do you think about it?

Reply:
This change leads to a regression on 32 bit systems with files larger than 4 GB. Just casting to size_t is a very bad idea when handling file sizes.

Reply:
Dropped the `size_t` cast of the file size.